controllers/djob_controller_databricks.go

```diff
@@ -88,13 +92,44 @@ func (r *DjobReconciler) delete(instance *databricksv1alpha1.Djob) error {
 
 	jobID := instance.Status.JobStatus.JobID
 
+	_, err := getJob(r, jobID)
+
 	// Check if the job exists before trying to delete it
-	if _, err := r.APIClient.Jobs().Get(jobID); err != nil {
+	if err != nil {
 		if strings.Contains(err.Error(), "does not exist") {
 			return nil
 		}
 		return err
 	}
 
-	return r.APIClient.Jobs().Delete(jobID)
+	return trackExecutionTime(djobDeleteDuration, func() error {
+		return r.APIClient.Jobs().Delete(jobID)
+	})
+}
+
+func getJob(r *DjobReconciler, jobID int64) (job models.Job, err error) {
+	return trackJob(djobGetDuration, djobGetSuccess, djobGetFailure, func() (models.Job, error) {
+		return r.APIClient.Jobs().Get(jobID)
+	})
+}
+
+func createJob(r *DjobReconciler, instance *databricksv1alpha1.Djob) (job models.Job, err error) {
+	return trackJob(djobCreateDuration, djobCreateSuccess, djobCreateFailure, func() (models.Job, error) {
+		return r.APIClient.Jobs().Create(*instance.Spec)
+	})
+}
+
+func trackJob(duration prometheus.Histogram, success prometheus.Counter, failure prometheus.Counter, 
+	f func() (models.Job, error)) (job models.Job, err error) {
+		job, err = trackJobExecutionTime(duration, func() (models.Job, error) {
+			return f()
+		})
+	
+		if err != nil {
+			failure.Inc()
+		} else {
+			success.Inc()
+		}
+	
+		return
```

Comment:
I have a preference for:
```suggestion
		return job, err
```

Reply:
awwwww i thought the game with Golang was to write the least, but most complicated, code you could 😂